Rubberduck.Deployment.Build/RubberduckPostBuildTask.cs

```diff
@@ -0,0 +1,365 @@
+            sb.AppendLine($"     IncludesDir: {IncludeDir}");
+            sb.AppendLine($"  FilesToExtract: {FilesToExtract}");
+            sb.AppendLine(string.Empty);
+
+            return sb.ToString();
+        }
+
+        private bool SetVCToolsPath()
+        {
+            var configuration = new SetupConfiguration();
+            var enumInstances = configuration.EnumInstances();
+            int fetched;
+            var instances = new ISetupInstance[1];
+            do
+            {
+                enumInstances.Next(1, instances, out fetched);
+
+                if (fetched <= 0)
+                {
+                    continue;
+                }
+
+                var instance = (ISetupInstance2) instances[0];
+                var packages = instance.GetPackages();
+                foreach (var package in packages)
+                {
+                    if (package.GetId() != "Microsoft.VisualStudio.Component.VC.Tools.x86.x64")
+                    {
+                        continue;
+                    }
+
+                    _rootPath = instance.ResolvePath();
+                    _batchPath = Directory.GetFiles(_rootPath, "VsDevCmd.bat", SearchOption.AllDirectories)
+                        .FirstOrDefault();
+
+                    return true;
+                }
+            } while (fetched > 0);
+
+            return false;
+        }
+
+        private void CleanOldImports(string dir)
+        {
+            this.LogMessage("Cleaing out old imports");
+
+            var files = Directory.GetFiles(dir, "DebugRegistryEntries.reg.imported_*.txt").OrderByDescending(f => f);
+            var i = 0;
+            foreach (var file in files)
+            {
+                if (i > 10)
+                {
+                    this.LogMessage($"Deleting {file}");
+                    File.Delete(file);
+                }
+
+                i++;
+            }
+        }
+
+        private void UpdateAddInRegistration()
+        {
+            this.LogMessage("Updating addin registration...");
+            var addInRegFile = Path.Combine(Path.GetDirectoryName(RegFilePath), "RubberduckAddinRegistry.reg");
+            var command = $"reg.exe import \"{addInRegFile}";
+            ExecuteTask(command, ignoreErrors:true);
+        }
+
+        private void ProcessDll(string file)
+        {
+            this.LogMessage($"Processing {file}...");
+
+            var parameters = new DllFileParameters(file, SourceDir, TargetDir);
+            
+            if (_batchPath == string.Empty)
+            {
+                this.LogMessage("Compiling with tlbexp...");
+                CompileWithTlbExp(parameters);
+            }
+            else
+            {
+                this.LogMessage("Compiling with midl...");
+                CreateIdlFile(parameters);
+                CompileWithMidl(parameters);
+            }
+
+            this.LogMessage("Extracting metadata using WiX...");
+            HarvestMetadataWithWix(parameters);
```

Comment:
I'd rename to `HarvestMetadataToFile`. That explains why there's no return value here.